students/TinaB/lesson08/circle.py

```diff
@@ -0,0 +1,123 @@
+#!/usr/bin/env python3
+"""
+Lesson 08 Assignment - Create a Circle Class
+"""
+from math import pi
+
+
+class Circle:
+    """Class for Circle - STEP ONE"""
+
+    def __init__(self, radius):
+        if radius is not None and radius < 0:
+            raise ValueError('Radius must be more than zero')
+
+        self._radius = radius
+        self._diameter = 2 * radius
+        self._area = pi * radius**2
+        self._perimeter = 2 * self._radius * pi
```

Comment:
Calculated values (diameter, radius, perimeter) should be handled in the getters and setters. Radius is best defined as as an instance attribute